src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -337,7 +337,8 @@ create_topic_properties_update(
                   update.properties.leaders_preference,
                   cfg.value,
                   kafka::config_resource_operation::set,
-                  noop_validator<config::leaders_preference>{},
+                  leaders_preference_validator{
+                    ctx.feature_table().local().should_sanction()},
```

Comment:
The same comment as on Ben's PR applies to this one as well: https://github.com/redpanda-data/redpanda/pull/23895#discussion_r1824295162

I think we should only fail validation if the request is both changing the value of the leaders preference (the final value won't be the same as the existing value) and it's changing to a restricted value (and we're in sanctioning mode).